web_infrastructure/jenkins_plugin.py

```diff
@@ -408,11 +408,18 @@ def _get_installed_plugins(self):
 
     def install(self):
         changed = False
+
         plugin_file = (
             '%s/plugins/%s.jpi' % (
                 self.params['jenkins_home'],
                 self.params['name']))
 
+        hpi_file = '%s/plugins/%s.hpi' % (
+            self.params['jenkins_home'],
+            self.params['name'])
+        if os.path.isfile(hpi_file):
+            os.rename(hpi_file, hpi_file+'.bak')
```

Comment:
I would say that if we managed to download the new `.jpi` file, we should be fine to delete the `.hpi`. I want to prevent the module to create clutter in the plugin directory.